### PR TITLE
feat(cdk/tree): expand CdkTree API to include methods previously attached to TreeControl

### PR DESCRIPTION
**Pull request by BobobUnicorn**

### CODE REVIEW

File: src/cdk/tree/tree.ts [hunk 1]
```diff
@@ -397,17 +478,11 @@ export class CdkTreeNode<T, K = T> implements FocusableOption, OnDestroy, OnInit
   }
   protected _data: T;
 
-  get isExpanded(): boolean {
-    return this._tree.treeControl.isExpanded(this._data);
-  }
-
   get level(): number {
     // If the treeControl has a getLevel method, use it to get the level. Otherwise read the
```
Comment (andrewseguin) [inline src/cdk/tree/tree.ts:482]: Revise comment about the tree control?

File: src/cdk/tree/nested-node.ts [hunk 2]
```diff
@@ -69,10 +69,10 @@ export class CdkNestedTreeNode<T, K = T>
 
   ngAfterContentInit() {
     this._dataDiffer = this._differs.find([]).create(this._tree.trackBy);
-    if (!this._tree.treeControl.getChildren && (typeof ngDevMode === 'undefined' || ngDevMode)) {
+    if (!this._tree.treeControl?.getChildren && (typeof ngDevMode === 'undefined' || ngDevMode)) {
       throw getTreeControlFunctionsMissingError();
     }
-    const childrenNodes = this._tree.treeControl.getChildren(this.data);
+    const childrenNodes = this._tree.treeControl!.getChildren(this.data);
```
Comment (andrewseguin) [inline src/cdk/tree/nested-node.ts:75]: It's not intuitive to me why on line 72 you have `treeControl?` and then you have `treeControl!`.

File: src/cdk/tree/tree.ts [hunk 3]
```diff
@@ -154,15 +159,32 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
     end: Number.MAX_VALUE,
   });
 
+  // TODO(cassc): will be used in a future PR
```
Comment (andrewseguin) [inline src/cdk/tree/tree.ts:162]: It's a little tricky to understand the use of this without seeing it in the current PR - in future PRs it'd be easier if we delay adding variables until they are used

**Review by andrewseguin — APPROVED** [review]

File: src/cdk/tree/tree.ts (hunk 3 above)
Comment (BobobUnicorn) [reply src/cdk/tree/tree.ts:162]: Makes sense, will do.

File: src/cdk/tree/nested-node.ts (hunk 2 above)
Comment (BobobUnicorn) [reply src/cdk/tree/nested-node.ts]: Hopefully the else if makes it more understandable?

File: src/cdk/tree/tree.ts (hunk 1 above)
Comment (BobobUnicorn) [reply src/cdk/tree/tree.ts]: Done!

**Review by andrewseguin — APPROVED** [review]

**Pull request merged**